src/mlpack/core/data/save_impl.hpp

```diff
@@ -381,20 +380,11 @@ bool Save(const std::vector<std::string>& files,
     throw std::runtime_error(oss.str());
     return false;
   }
-  // We transpose by default. So, un-transpose if necessary.
-  if (!transpose)
-    matrix = arma::trans(matrix);
 
-  arma::Mat<unsigned char> img;
-  bool status = Save(files[0], img, info, fatal, transpose);
-
-  // Decide matrix dimension using the image height and width.
-  matrix.set_size(info.Width() * info.Height() * info.Channels(), files.size());
-  matrix.col(0) = img;
-
-  for (size_t i = 1; i < files.size() ; i++)
+  bool status = true;
+  for (size_t i = 0; i < files.size() ; i++)
   {
-    arma::Mat<unsigned char> colImg(matrix.colptr(i), matrix.n_rows, 1,
+    arma::Mat<eT> colImg(matrix.colptr(i), matrix.n_rows, 1,
         false, true);
     status &= Save(files[i], colImg, info, fatal, transpose);
```

Comment:
umm , should we ?

As of now out of many files, one may be corrupt and doesn't save but other do, If false then it will save till first corrupt file, right ?

Reply:
Hmm, up to you.  I can see it as being okay both ways.